crates/nu-protocol/src/eval_const.rs

```diff
@@ -508,6 +508,14 @@ impl Eval for EvalConst {
         block_id: BlockId,
         span: Span,
     ) -> Result<Value, ShellError> {
+        // If parsing errors exist in the subexpression, don't bother to evaluate it.
+        if working_set
+            .parse_errors
+            .iter()
+            .any(|error| span.contains_span(error.span()))
+        {
+            return Err(ShellError::ParseErrorInConstant { span });
```

Comment:
Seems like an interim solution to me that this leeds to another error instead of being able to forgo if he parsing does not succeed.

I hope most `ParseErrors` are spanned correctly.